### PR TITLE
chore(ci): use `docker/metadata-action`

### PR DESCRIPTION
**Pull request by rmoreas**
Replaces custom `setup-image-vars` action by `docker/metadata-action` and automatically add latest tag on semver tag.

### CODE REVIEW

**Comment by yxxhero** [comment]
@rmoreas please fix DCO issue.

**Comment by yxxhero** [comment]
image

**Comment by rmoreas** [comment]
> image

That is intended: currently tags are only generated on semver git tags and push to main (canary). But we can add tags for other events to if desired.

**Comment by yxxhero** [comment]
we can remove ./.github/actions/setup-image-vars if this PR be merged.

**Comment by rmoreas** [comment]
> we can remove ./.github/actions/setup-image-vars if this PR be merged.

Yes, this will not be needed anymore

**Comment by yxxhero** [comment]
@rmoreas new logic is better.

**Comment by rmoreas** [comment]
If you like, I can remove `.github/actions/setup-image-vars` in this PR...

**Comment by yxxhero** [comment]
@rmoreas just do it.

File: .github/workflows/images.yaml [hunk 1]
```diff
@@ -65,10 +73,10 @@ jobs:
         uses: docker/build-push-action@v3
         with:
           context: .
-          file: ${{ matrix.dockerfile }}
+          file: ${{ matrix.image.dockerfile }}
           platforms: linux/amd64,linux/arm64
           push: ${{ github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/tags/') }}
-          tags: |
-            ghcr.io/${{ github.repository }}${{ steps.vars.outputs.suffix }}:${{ steps.vars.outputs.tag }}
+          tags: ${{ steps.meta.outputs.tags }}
```
Comment (yxxhero) [inline .github/workflows/images.yaml:79]: what's `steps.meta.outputs.tags`, maybe less `ghcr.io/....`?

File: .github/workflows/images.yaml [hunk 2]
```diff
@@ -65,10 +73,10 @@ jobs:
...
+          labels: ${{ steps.meta.outputs.labels }}
```
Comment (yxxhero) [inline .github/workflows/images.yaml:80]: remove extra space.

**Comment by yxxhero** [comment]
last step: rm .github/actions/setup-image-vars

File: .github/workflows/images.yaml (hunk 1 above)
Comment (rmoreas) [reply .github/workflows/images.yaml:79]: `steps.meta.outputs.tags` includes `ghcr.io/...`. The image repos is set by the `meta` step using the `with.images` setting as input.

**Review by yxxhero — APPROVED** [review]
awesome work.

**Pull request merged**